### PR TITLE
add args to conversion to be returnable as json

### PR DESCRIPTION
**Pull request by JL-Brothers**
Closes Issue #698

### CODE REVIEW

File: openc3/lib/openc3/conversions/conversion.rb [hunk 1]
```diff
@@ -67,6 +70,7 @@ def as_json(*_a)
       result['converted_type'] = @converted_type if @converted_type
       result['converted_bit_size'] = @converted_bit_size if @converted_bit_size
       result['converted_array_size'] = @converted_array_size if @converted_array_size
+      result['args'] = JSON.generate(@args) if @args
```
Comment (ryanmelt) [inline openc3/lib/openc3/conversions/conversion.rb:73]: as_json shouldn't call JSON.generate.  The goal is return objects that are ready to be converted to JSON, but not return an actual JSON string yet.   Change to "= @params.as_json(*a)" and remove underscore from the existing _a

File: openc3/python/openc3/conversions/conversion.py [hunk 2]
```diff
@@ -61,4 +63,6 @@ def as_json(self):
             result["converted_bit_size"] = self.converted_bit_size
         if self.converted_array_size is not None:
             result["converted_array_size"] = self.converted_array_size
+        if self.args is not None:
+            result["args"] = json.dumps(self.args)
```
Comment (ryanmelt) [inline openc3/python/openc3/conversions/conversion.py:67]: Same comment as in Ruby.

File: openc3/templates/conversion/conversion.py [hunk 3]
```diff
@@ -17,6 +17,7 @@ def __init__(self):
     #   a string, integer, float, or array of values.
     # @param packet [Packet] The packet object where the conversion is defined
     # @param buffer [String] The raw packet buffer
+    # @param args [Array] The arguments to return
```
Comment (ryanmelt) [inline openc3/templates/conversion/conversion.py:20]: args is not an parameter to call.  remove.

File: openc3/lib/openc3/conversions/conversion.rb [hunk 4]
```diff
@@ -30,12 +30,15 @@ class Conversion
     attr_reader :converted_bit_size
     # @return [Integer] The size in bits of the converted array value
     attr_reader :converted_array_size
+    # @return [] The Array of arguments passed in
+    attr_reader :args
```
Comment (ryanmelt) [inline openc3/lib/openc3/conversions/conversion.rb:34]: Probably rename to params.  If you look at the as_json method in a lot of our existing conversion classes they already are reporting their arguments as 'params'

File: openc3/templates/conversion/conversion.rb [hunk 5]
```diff
@@ -12,6 +12,8 @@ def initialize
       # Size of the converted type in bits
       # Use 0 for :STRING or :BLOCK where the size can be variable
       @converted_bit_size = 0
+      # return the arguments used
+      @args = []
```
Comment (ryanmelt) [inline openc3/templates/conversion/conversion.rb:16]: This should default to nil, like in the base class.

**Review by ryanmelt — CHANGES_REQUESTED** [review]
Need look through all the existing conversion classes and if they take arguments make sure the @params variable gets filled with those arguments.

If possible remove subclass as_json methods where they would do the same thing as the base class if @params is filled out.

File: openc3/lib/openc3/conversions/processor_conversion.rb [hunk 6]
```diff
@@ -40,6 +40,7 @@ def initialize(processor_name, result_name, converted_type = nil, converted_bit_
       end
       @converted_bit_size = Integer(converted_bit_size) if ConfigParser.handle_nil(converted_bit_size)
       @converted_array_size = Integer(converted_array_size) if ConfigParser.handle_nil(converted_array_size)
+      @params = [@processor_name, @result_name]
```
Comment (ryanmelt) [inline openc3/lib/openc3/conversions/processor_conversion.rb:43]: MIssing converted_ params

File: openc3/python/openc3/conversions/conversion.py [hunk 7]
```diff
@@ -25,12 +25,14 @@ class Conversion:
     # attr_reader :converted_bit_size
     # # self.return [Integer] The size in bits of the converted array value
     # attr_reader :converted_array_size
+    # attr_reader :args
```
Comment (ryanmelt) [inline openc3/python/openc3/conversions/conversion.py:28]: Should be params

File: openc3/python/openc3/conversions/processor_conversion.py [hunk 8]
```diff
@@ -45,6 +45,7 @@ def __init__(
             self.converted_bit_size = int(converted_bit_size)
         if ConfigParser.handle_none(converted_array_size):
             self.converted_array_size = int(converted_array_size)
+        self.params = [self.processor_name, self.result_name]
```
Comment (ryanmelt) [inline openc3/python/openc3/conversions/processor_conversion.py:48]: Missing converted_ params

File: openc3/templates/conversion/conversion.py (hunk 3 above)
Comment (ryanmelt) [inline openc3/templates/conversion/conversion.py:20]: Remove args

**Review by ryanmelt — CHANGES_REQUESTED** [review]

File: openc3/lib/openc3/conversions/processor_conversion.rb (hunk 6 above)
Comment (jmthomas) [reply openc3/lib/openc3/conversions/processor_conversion.rb:43]: Base class records those ... they are output in the `to_config` method

File: openc3/python/openc3/conversions/processor_conversion.py (hunk 8 above)
Comment (jmthomas) [reply openc3/python/openc3/conversions/processor_conversion.py:48]: Base class records those ... they are output in the to_config method. as_json outputs them separately and the remaining parameters get put into params.

**Comment by ryanmelt** [comment]
This ticket still needs to capture all params.

**Review by ryanmelt — APPROVED** [review]

**Pull request merged**